pkg/spdk_test.go

```diff
@@ -396,16 +802,51 @@ func checkReplicaSnapshots(c *C, spdkCli *client.SPDKClient, replicaList []strin
 			c.Assert(snap, NotNil)
 
 			for _, childSnapName := range childrenList {
-				if childSnapName == "volume-head" {
-					c.Assert(snap.Children[replicaName], Equals, true)
-				} else {
-					childLvolName := server.GetReplicaSnapshotLvolName(replicaName, childSnapName)
-					c.Assert(snap.Children[childLvolName], Equals, true)
+				c.Assert(snap.Children[childSnapName], Equals, true)
+				if childSnapName != types.VolumeHead {
 					childSnap := replica.Snapshots[childSnapName]
 					c.Assert(childSnap, NotNil)
-					c.Assert(childSnap.Parent, Equals, server.GetReplicaSnapshotLvolName(replicaName, snapName))
+					c.Assert(childSnap.Parent, Equals, snapName)
 				}
 			}
 		}
 	}
 }
+
+func revertSnapshot(c *C, spdkCli *client.SPDKClient, snapshotName, volumeName, engineName string, replicaAddressMap map[string]string) {
+	engine, err := spdkCli.EngineGet(engineName)
+	c.Assert(err, IsNil)
+
+	if engine.State != types.InstanceStateRunning {
+		return
+	}
+
+	prevFrontend := engine.Frontend
+	prevEndpoint := engine.Endpoint
+	if prevFrontend != types.FrontendEmpty {
+		// Restart the engine without the frontend
+		err = spdkCli.EngineDelete(engineName)
+		c.Assert(err, IsNil)
+		engine, err = spdkCli.EngineCreate(engineName, volumeName, types.FrontendEmpty, defaultTestLvolSize, replicaAddressMap, 1)
+		c.Assert(err, IsNil)
+		c.Assert(engine.State, Equals, types.InstanceStateRunning)
+		c.Assert(engine.ReplicaAddressMap, DeepEquals, replicaAddressMap)
+		c.Assert(engine.Port, Equals, int32(0))
+		c.Assert(engine.Endpoint, Equals, "")
+	}
+
+	err = spdkCli.EngineSnapshotRevert(engineName, snapshotName)
```

Comment:
You added test cases of snapshot and revert. All passed?

Reply:
They are added by @shuo-wu and all passed.

Reply:
Like the tests, quite solid.